### PR TITLE
feat(cli): add go support for provider add

### PR DESCRIPTION
**Pull request by DanielMSchmidt**

### CODE REVIEW

File: packages/cdktf-cli/lib/dependencies/dependency-manager.ts [hunk 1]
```diff
@@ -120,17 +121,18 @@ export class ProviderConstraint {
...
   ) {
-    this.packageManager =
-      targetLanguage !== Language.GO
-        ? PackageManager.forLanguage(targetLanguage, this.projectDirectory)
-        : undefined;
+    this.packageManager = PackageManager.forLanguage(
+      targetLanguage,
+      this.projectDirectory
+    );
+    undefined;
```
Comment (ansgarm) [inline packages/cdktf-cli/lib/dependencies/dependency-manager.ts:135]:
```suggestion
```

File: packages/cdktf-cli/lib/dependencies/dependency-manager.ts [hunk 2]
```diff
@@ -249,10 +242,11 @@ export class DependencyManager {
...
   private convertPackageName(name: string): string {
     switch (this.targetLanguage) {
-      case Language.GO:
-        throw Errors.Internal(
-          "pre-built providers are not supported for Go yet"
-        );
+      case Language.GO: // e.g. github.com/hashicorp/cdktf-provider-opentelekomcloud-go
+        return `github.com/hashicorp/cdktf-provider-${name.replace(
+          "@cdktf/provider-",
+          ""
```
Comment (ansgarm) [inline packages/cdktf-cli/lib/dependencies/dependency-manager.ts:248]: package name requires a `/${providerName}` at the end. See [here](https://github.com/hashicorp/terraform-cdk/issues/723#issuecomment-1189473189).
So, maybe pull the name into a variable and use it twice?

**Review by ansgarm — COMMENTED** [review]
Small thing to fix, and we should update the [test](https://github.com/hashicorp/terraform-cdk/blob/main/test/go/provider-add-command/test.ts) for Go that currently only check locally added providers I think. The tests for the other languages might offer some inspiration

**Comment by DanielMSchmidt** [comment]
Added an issue to test this functionality: https://github.com/hashicorp/terraform-cdk/issues/1945

File: packages/cdktf-cli/lib/dependencies/dependency-manager.ts [hunk 3]
```diff
@@ -248,25 +240,18 @@ export class DependencyManager {
    * Converts an NPM package name of a pre-built provider package to the name in the target language
    */
   private convertPackageName(name: string): string {
+    const providerName = name.replace("@cdktf/provider-", "");
     switch (this.targetLanguage) {
-      case Language.GO:
-        throw Errors.Internal(
-          "pre-built providers are not supported for Go yet"
-        );
+      case Language.GO: // e.g. github.com/hashicorp/cdktf-provider-opentelekomcloud-go
```
Comment (ansgarm) [inline packages/cdktf-cli/lib/dependencies/dependency-manager.ts:245]:
```suggestion
      case Language.GO: // e.g. github.com/hashicorp/cdktf-provider-opentelekomcloud-go/opentelekomcloud
```

**Review by ansgarm — APPROVED** [review]
Nice 👏

**Pull request merged**